File: src/Tokenization/token.rb

# literal and text are the same thing, replace "text" with "literal" in entire code
class Token
    def initialize(tok_type, text, literal, line, filename)
        @tok_type = tok_type
        @text = text
        @literal = text 
        @line = line 
        @filename = filename
    end

    def print
        return "\n filename: #{@filename}\n line: #{(@line).to_s}\n token type: #{@tok_type}\n literal: #{@literal}\n\n\n"
    end

    def getText
        return @literal
    end

    def getFilename
        return @filename
    end

    def getLine
        return @line.to_s()
    end

    def getType
        return @tok_type
    end

end

File: src/Tokenization/token.py

class Token:
    def __init__(self, type_symbol: str, literal: str, file_name: str, line_number: int, column_number: int):
        self.type_symbol = type_symbol
        self.literal = literal
        self.file_name = file_name
        self.line_number = line_number
        self.column_number = column_number

    def get_type(self) -> str:
        return self.type_symbol
    
    def get_name(self):
        return self.literal
    
    def set_type(self, type_symbol: str) -> None:
        self.type_symbol = type_symbol

    def to_string(self) -> str:
        repr = "Token:\n"
        repr += " file: " + self.file_name + "\n"
        repr += " line number: " + str(self.line_number) + "\n"
        repr += " column number: " + str(self.column_number) + "\n"
        repr += " literal: " + self.literal + "\n"
        repr += " type: " + self.type_symbol + "\n"
        return repr
    

    def __str__(self) -> str:
        return self.to_string()
    
